projects/plugins/jetpack/class.json-api.php

```diff
@@ -202,12 +202,17 @@ public function add( WPCOM_JSON_API_Endpoint $endpoint ) {
 	}
 
 	/**
-	 * Determine if a string is truthy.
+	 * Determine if a string is truthy. If it's not a string, which can happen with
+	 * not well-formed data coming from Jetpack sites, we still consider it a truthy value.
 	 *
 	 * @param string $value "1", "t", and "true" (case insensitive) are falsey, everything else isn't.
 	 * @return bool
 	 */
 	public static function is_truthy( $value ) {
+		if ( ! is_string( $value ) ) {
```

Comment:
Haven't backtraced the usage of this, maybe you can clarify. But adding this would mean that values such as `0`, `null` and `false` would return `true` here. Is this what we want?

Reply:
Nod, this would change the behavior for those particular things, which I would think we should try to maintain. These two functions, `is_truthy` and the `is_falsy` below are interestingly setup (only expecting "1", "t", "true" as strings (thus excluding bool `true` and the number 1) and false only expecting strings as well.

Overall, I'd like to understand better (just driving by the comment right now) why these functions exist and why they're so narrow-scoped on what's eligible for true/false determination. Maybe they should be depreciated for `is_string_truthy`, etc :) with a doing_it_wrong if a non-string is passed to it? Something that would raise a flag that something unexpected is happening without the fatal.

Also, let's update `is_falsy` below since it could suffer from the same fatal for non-string elements being passed to it.

Perhaps another way could be a try/catch for the casting to a string. We can catch the type of fatal we say for a non-convertible type and return a doing_it_wrong, while allowing for convertible other types through.

Reply:
I note an odd edge case, `is_truthy( true )` would work because `(string) true` => `'1'`, but `is_falsy( false )` would not because `(string) false` => `''` and the empty string is not considered falsey by `is_falsy`.

I see `is_falsy` was added in 9ed52fc36e1c124a591dca48a8daddf0d4d9bfb1 to do something like `if ( is_falsy( $value ) ) { $value = false }`, handling cases where someone supplied "false" to the API parameter. Which probably explains the oversight.

`is_truthy` came in a big merge in d47c5440fb0c1c8a9df44b13f89f357aecebf370 and seems to have been used like `(bool) $value` there to only accept certain strings as "true".

The doc block specifying "string" was added much later, in #22322, and I may have guessed wrong at what exactly might be passed in.

Reply:
I have updated both methods, thank you all!
@anomiex that's a great find, but I think it's a separate fix, let's not include it into this PR.

Reply:
@anomiex it looks like your suggestion will actually fix the failing tests, I'm going to try it now.